reporter/src/main/kotlin/utils/FreemarkerTemplateProcessor.kt

```diff
@@ -0,0 +1,242 @@
+            }
+
+        /**
+         * Return licenses that are configured to require a
+         * [source code offer in a notice file][License.includeSourceCodeOfferInNoticeFile] in the
+         * [LicenseConfiguration].
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun filterIncludeSourceCodeOfferInNoticeFile(licenses: Collection<ResolvedLicense>): List<ResolvedLicense> =
+            licenses.filter { resolvedLicense ->
+                licenseConfiguration.licenses.find { it.id == resolvedLicense.license }
+                    ?.includeSourceCodeOfferInNoticeFile ?: false
+            }
+
+        /**
+         * Return a [LicenseView] constant by name to make them easily available to the Freemarker templates.
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun licenseView(name: String): LicenseView =
+            LicenseView.Companion::class.memberProperties
+                .first { it.name == name }
+                .get(LicenseView.Companion) as LicenseView
+
+        /**
+         * Merge the [ResolvedLicense]s of multiple [models] and filter them using [licenseView]. [Omits][omitExcluded]
+         * excluded packages, licenses, and copyrights by default. The returned list is sorted by license identifier.
+         */
+        @JvmOverloads
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun mergeLicenses(
+            models: Collection<PackageModel>,
+            licenseView: LicenseView = LicenseView.ALL,
+            omitExcluded: Boolean = true
+        ): List<ResolvedLicense> =
+            mergeResolvedLicenses(
+                models.filter { !omitExcluded || !it.excluded }.flatMap {
+                    val licenses = it.license.filter(licenseView).licenses
+                    if (omitExcluded) licenses.filterExcluded() else licenses
+                }
+            )
+
+        /**
+         * Return a list of [ResolvedLicense]s where all duplicate entries for a single license in [licenses] are
+         * merged. The returned list is sorted by license identifier.
+         */
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun mergeResolvedLicenses(licenses: List<ResolvedLicense>): List<ResolvedLicense> =
+            licenses.groupBy { it.license }
+                .map { (_, licenses) -> licenses.merge() }
+                .sortedBy { it.license.toString() }
+    }
+}
+
+private fun List<ResolvedLicense>.merge(): ResolvedLicense {
+    require(isNotEmpty()) { "Cannot not merge an empty list." }
+    return ResolvedLicense(
+        license = first().license,
+        sources = flatMapTo(mutableSetOf()) { it.sources },
+        originalDeclaredLicenses = flatMapTo(mutableSetOf()) { it.originalDeclaredLicenses },
+        locations = flatMapTo(mutableSetOf()) { it.locations }
+    )
+}
```

Comment:
I'd prefer variant 2 for less nesting.

Reply:
I just realized variant 2 causes an issue with file naming now. Due to code

    val outputFile = outputDir.resolve("$NOTICE_FILE_PREFIX$id")

we would now get e.g. `NOTICE_notice-default` instead of `NOTICE_default` as the file name. How should be solve this? By going for variant 1, or by changing the logic to deduce the output name? Personally, I'd have a slight preference for the former.

Reply:
Variant 1 is possible. The `FreemarkerTemplateProcessor` then needs an additional field, i.e., the folder where it shall search for the template ids.

Reply:
Yeah, let's go for that approach.